src/models/lineChart.js

```diff
@@ -191,7 +191,19 @@ nv.models.lineChart = function() {
                         margin.top = legend.height();
                         availableHeight1 = nv.utils.availableHeight(height, container, margin) - (focusEnable ? focusHeight : 0);
                     }
+                
+                //Do not display the legend if there isn't enough space
+                if (availableHeight1 === 0) {
+                    g.select('.nv-legendWrap')
+                        .datum([])
+                        .call(legend);
 
+                    if (margin.top != legend.height()) {
+                        margin.top = legend.height();
+                        availableHeight1 = nv.utils.availableHeight(height, container, margin) - (focusEnable ? focusHeight : 0);
+                    }
```

Comment:
we just had a series of PR's to fix it so that it wouldn't overwrite the margin randomly like this...  should probably try to use the same method.